server/src/main/java/org/elasticsearch/env/DefaultBuildVersion.java

```diff
@@ -37,6 +37,11 @@ final class DefaultBuildVersion extends BuildVersion {
         this.version = Version.fromId(versionId);
     }
 
+    DefaultBuildVersion(String version) {
+        this.version = Version.fromString(Objects.requireNonNull(version));
+        this.versionId = this.version.id();
```

Comment:
why store this, the use in `id()` could call `version.id()`?

Reply:
Oh yes!